Lua/tests/syntax_test_lua.lua

```diff
@@ -0,0 +1,491 @@
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+
+    foo.baz [=[ a string ]=];
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+
+    foo.baz {};
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+--          ^^ meta.function-call.arguments meta.mapping
+
+    foo[bar + baz];
+--     ^^^^^^^^^^^ meta.brackets
+--     ^ punctuation.section.brackets.begin
+--               ^ punctuation.section.brackets.end
+
+--FUNCTION CALLS
+
+    f(42);
+--  ^ variable.function
+--   ^ meta.function-call.arguments meta.group
+--   ^ punctuation.section.group.begin
+--      ^ punctuation.section.group.end
+
+    f "argument";
+--  ^ variable.function
+--    ^^^^^^^^^^ meta.function-call.arguments string.quoted.double
+
+    f
+    'argument';
+--  ^^^^^^^^^^ meta.function-call.arguments string.quoted.single
+
+    f [[ foo ]];
+--  ^ variable.function
+--    ^^^^^^^^^ meta.function-call.arguments string.quoted.block
+
+    f {};
+--  ^ variable.function
+--    ^^ meta.function-call.arguments meta.mapping
+
+--FUNCTIONS
+
+    function function_name( a, b, ... )
+--  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--  ^^^^^^^^ storage.type.function
+--           ^^^^^^^^^^^^^ entity.name.function
+--                        ^^^^^^^^^^^^^ meta.group
+--                        ^ punctuation.section.group.begin
+--                          ^ variable.parameter.function
+--                           ^ punctuation.separator.comma
+--                             ^ variable.parameter.function
+--                              ^ punctuation.separator.comma
+--                                ^^^ constant.language
+--                                    ^ punctuation.section.group.end
+    end
+--  ^^^ meta.function meta.block keyword.control.end
+
+    function foo.bar:baz (...) end
+--  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--           ^^^ variable.other - entity
+--              ^ punctuation.accessor
+--               ^^^ variable.other - entity
+--                  ^ punctuation.accessor
+--                   ^^^ entity.name.function
+--                       ^^^^^ meta.group
+--                             ^^^ keyword.control.end
+
+    function foo
+        .bar () end
+--      ^^^^^^^^^^^ meta.function
+--      ^ punctuation.accessor
+--       ^^^ entity.name.function
+
+    local function foo () end
+--  ^^^^^ storage.modifier
+--        ^^^^^^^^^^^^^^^^^^^ meta.function
+--                 ^^^ entity.name.function
+
+    ~function( a, b, ... ) end;
+--   ^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--   ^^^^^^^^ storage.type
+--           ^^^^^^^^^^^^^ meta.group
+--                         ^^^ keyword.control.end
+
+    ~function foo () end;
+--   ^^^^^^^^^^^^^^^^^^^ meta.function
+--            ^^^ entity.name.function
+
+    foo = function() end;
+--  ^^^ entity.name.function
+
+    foo.bar = function() end;
+--      ^^^ entity.name.function
+
+--STATEMENTS
+
+    end;
```

Comment:
Why apply the keyword scope as well? For color schemes that don't set a background color on `invalid.illegal`, this will look like a normal keyword.

Reply:
Fixed.